app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/completeregistration/CompleteRegistrationViewModel.kt

```diff
@@ -215,13 +219,7 @@ class CompleteRegistrationViewModel @Inject constructor(
             }
 
             is RegisterResult.Success -> {
-                mutableStateFlow.update { it.copy(dialog = null) }
-                sendEvent(
-                    CompleteRegistrationEvent.NavigateToLogin(
-                        email = state.userEmail,
-                        captchaToken = registerAccountResult.captchaToken,
-                    ),
-                )
+                trySendAction(Internal.AttemptLogin(registerAccountResult.captchaToken))
```

Comment:
Why exactly do we need an internal action triggered here? Why not just run the logic that happens in `handleLoginAttempt` here?

Reply:
happy to un-Matryoshka doll this, I was trying to follow the pattern of the other "Receive*Result"

Reply:
actually I guess this would maybe be a reason: https://github.com/bitwarden/android/pull/3842#discussion_r1736454964

cc: @david-livefront can you take a look at the previous commit and lmk if that would be better

Reply:
I agree with Brian here, the pattern is usually to launch a scope to do the work and send an internal action to process the result (which usually updates the state).

Reply:
wow I need to work on my reading comprehension, good show lads, ty for the patience.